agent.go

```diff
@@ -665,6 +665,12 @@ func (ep *endpoint) addServiceInfoToCluster(sb *sandbox) error {
 }
 
 func (ep *endpoint) deleteServiceInfoFromCluster(sb *sandbox, method string) error {
+
+	if !ep.serviceEnabled {
```

Comment:
it should be only part of the local store and in memory right ? This need not be propogate to the store. So Marshalling and Unmarshalling may not be needed ? getendpoints returns endpoints from in memory.